versioned_docs/version-2020fa/assignment5.md

```diff
@@ -0,0 +1,43 @@
+---
+id: assignment5
+title: Assignment 5
+---
+
+For A5, you'll be building a React App that utilizes conditional rendering
+and the concept of lifting state up.
+
+Sam is trying to build a system to search through his friends list on
+SamBook, a new social media platform that he built.
+
+For this search engine, you'll be entering a search query and
+maintaining two separate components: the first shows
+all results that contain the search query, and the
+second shows all OTHER results (i.e. all results that don't contain
+the search query).
+
+## Part 1
+
+In App.tsx, initialize a state variable that contains the search query.
+Also, create an input field that interacts with this state variable.
+
+Create a state variable that contains Sam's friends list. Make
+another input field and a button associated with it to add a name to
+Sam's list. Do not display the list in this component; we will be passing
+it down to child components.
```

Comment:
I think the following things should be more clearly specified:
- Type of friend
- We expect the students to hardcode something creative